chingu-voyages/v52-tier3-team-38
# Ross/feature/styling/appointment form

# Description
I styled the forms! It started out as styling the buttons for the Appointment Form, and then I used that to style the other buttons on the Sign Up form and the Log In form. From there, I used vanilla Bootstrap flex for the Log In form to get the layout just right. I was then able to take that and do the same with the Sign Up form! From there I was able to achieve my ultimate goal--styling the Appointment Form! This was a fun reminder on how to use flex in vanilla Bootstrap!

## Related to the following Jira/Trello issue 

- Resident Interface

## Main changes explained:
- Added cancel buttons to all forms for improved navigation back to either root home page or profile pages.
- Refactored login form layout using Bootstrap flexbox for better organization and responsiveness.
- Applied the same flexbox wrapper pattern to signup form for consistency.
- Enhanced appointment form layout with flexbox utilities throughout for improved spacing and responsiveness
- Standardized button styling using Bootstrap React's grid system

## How to test:
Check into current branch
Run npm install and npm run dev to run this PR locally
Test cancel button functionality:

Go to /login - verify cancel button returns to home page
Go to /signup - verify cancel button returns to home page
Go to /user/[_id]/submit-request - verify cancel button returns to user profile

Verify responsive layout:

Test all forms at different screen sizes
Confirm forms remain centered and properly spaced
Check that appointment form fields stack appropriately on mobile

Verify form functionality remains intact:

Test login with valid credentials
Test signup process
Test appointment submission

## Screenshots or videos of changes:
Appointment Request form.
![image](https://github.com/user-attachments/assets/736b9763-76ed-4f28-9f3b-b28483b72a64)
![image](https://github.com/user-attachments/assets/081ddd6e-be01-4b3f-842c-522523e57d95)


Sign Up form!
![image](https://github.com/user-attachments/assets/111d61df-ee0f-427a-b4e1-96b74da670ca)
![image](https://github.com/user-attachments/assets/32540524-dcb0-4799-b777-ce22e367d118)

Sign In form!
![image](https://github.com/user-attachments/assets/cbc787e9-e2cd-414d-a1c8-72b4df3a3787)
![image](https://github.com/user-attachments/assets/c8bc2e91-dd46-4a4e-b8a3-692ca134436f)


## Note:
- Now that we have the Flexbox down, we can use these flex wrappers for other pages, and then style from there! Should make styling less stressful!
- Forms maintain full functionality while providing better navigation.
- All styling uses Bootstrap's built-in utility classes (d-flex! flex!) for maintainability.
